test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ETags/DerivedETagTests.cs

```diff
@@ -91,5 +104,19 @@ private static IEdmModel GetEdmModel(WebRouteConfiguration configuration)
 
             Assert.True(jsonETags.FirstOrDefault() == singletonEtag, "Singleton has different etags than Set");
         }
+
+        [Fact]
+        public async Task DerivedEntitySetsHaveETagsTest()
+        {
+            string requestUri = this.BaseAddress + "/derivedEtag/ETagsDerivedCustomers?$select=Id";
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, requestUri);
+            request.Headers.Accept.ParseAdd("application/json");
+            HttpResponseMessage response = await this.Client.SendAsync(request);
+            Assert.True(response.IsSuccessStatusCode);
+            var jsonResult = await response.Content.ReadAsObject<JObject>();
+            var derivedEtags = jsonResult.GetValue("value").Select(e => e["@odata.etag"].ToString());
+
+            Assert.StartsWith("W/\"bnVsbA==", String.Concat(derivedEtags));
```

Comment:
Can we do derivedETags.Single() here?

Reply:
the response payload will include multiple etag's value. So, I can use First() to check.  #resolved.